source/ID/concordium-did.rst

```diff
@@ -0,0 +1,546 @@
+
+Smart Contract Instance DID
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A smart contract instance DID can be created by `deploying a smart contract module <deploy-module_>`_ and `initializing a smart contract instance <initialize-contract-instance_>`_ on the ``NET`` blockchain.
+The resulting DID is ``did:ccd:NET:sci:IND:SUBIND`` where ``IND``, ``SUBIND`` are the index and the subindex of the instance.
+
+Public Key Cryptography DID
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A public key cryptography DID can be created by generating a fresh Ed25519 key pair.
+The resulting DID is ``did:ccd:NET:pkc:PK`` where ``PK`` is the base58 encoded public key.
+These DIDs are not registered on the blockchain.
+
+Read
+----
+
+Account DID
+^^^^^^^^^^^
+
+The DID document information for a DID of the form
+
+``did:ccd:NET:acc:ADDR``
+
+can be resolved by looking up the account with address ``ADDR`` on blockchain ``NET``.
+
+Data required to construct the DID document can be acquired by using the gRPC interface command ``GetAccountInfo``.
+
+See the details in the `gRPC v2 documentation`_.
+
+From the command line, ``concordium-client`` allows to retrieve the data in the following way:
+
+.. code-block:: console
+
+    $concordium-client raw GetAccountInfo ADDR
+
+.. TODO update, once we have a DID resolver
+
+
+Smart Contract Instance DID
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The DID document information for a DID of the form
+
+``did:ccd:NET:sci:IND:SUBIND``
+
+can be resolved by looking up the smart contract instance with indices ``IND``, ``SUBIND`` on blockchain ``NET``.
+This includes a lookup of the owner's account.
+
+Data required to construct the DID document can be acquired by using the gRPC interface command ``GetInstanceInfo``.
+
+See the details in the `gRPC v2 documentation`_.
+
+From the command line, ``concordium-client`` allows for retrieving the data in the following way:
+
+.. code-block:: console
+
+  $concordium-client contract show IND
+
+.. TODO update, once we have a DID resolver
+
+
+.. _sci-reference:
+
+Smart Contract Instance Reference
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+*Dereferencing* the smart contract DID reference invokes the specified entrypoint.
+
+Dereferencing a DID reference of the form
+
+``did:ccd:NET:sci:IND:SUBIND/EP?parameter_json=PAR[&format=FMT]``
+
+can be done by using the gRPC interface command ``InvokeInstance``.
+The entrypoint is considered a *view*: no state changes are persisted, only the result of the invocation is returned to the caller.
+The parameter ``PAR`` is passed to the entrypoint.
+
+The result of the invocation is the return value produced by the contract or an error, if the invocation failed.
+The optional query parameter ``format=FMT`` specifies the formatting of the return value.
+If not specified, the return value is in the JSON format corresponding to the embedded smart contract schema.
+If specified, the ``FMT`` value can be the following:
```

Comment:
Is it an error if this is specified and the schema is not there?

I would generally prefer not to rely on any schemas. They are optional.

Reply:
After discussing the formatting of return results with @abizjak , the proposed solution is the following. Instead of having the `format` parameter, we introduce the `standard` parameter, which refers to a CIS standard associated with the smart contract. For each standard, we define how the return data should be rendered. Specifically for the smart contract registries, we specify that the entrypoints that return public keys a rendered as DID documents with the appropriate  `verificationMethod`.